docs/source/developer_guides/model_merging.md

```diff
@@ -0,0 +1,178 @@
+<!--Copyright 2024 The HuggingFace Team. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# Model merging
```

Comment:
One more comment, I am not sure if these readme guides/documentation typically provide citations to the original papers or not but that would be good to have the bottom of the page.

Reply:
Generally, we provide a link to the original paper that implements the method as a reference/citation for users who are more interested. I don’t think it’s as common outside of academia to have Citation sections in documentation. For user-facing guides like these, users are typically only interested in quickly copying/pasting the code to get it to run 🙂 

We'll add a link to the blog post @sayakpaul is working on once it's ready where it'll be more appropriate to acknowledge!